distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkAction.java

```diff
@@ -0,0 +1,305 @@
+
+        if(buildingMessage.getPeer() != null && buildingMessage.getPeer().isValid())
+
+            currentPeer = buildingMessage.getPeer();
+
+        if(parameteres.length == TOTAL_PARAMS){
+
+            int actionType;
+
+            try {
+
+                actionType = Integer.parseInt(parameteres[ACTION_POSITION]);
+
+            }
+
+            catch (NumberFormatException e){
+
+                throw new IllegalArgumentException(e.getMessage() + "\n" + ACTION_CODE_NOT_FOUND_ERROR_MSG);
+
+            }
+            int nodeId;
+            try {
+
+                nodeId = Integer.parseInt(parameteres[ID_POSITION]);
+
+            }
+
+            catch (NumberFormatException e) {
+
+                throw new IllegalArgumentException(e.getMessage() + "\n" + ID_NOT_FOUND_ERROR_MSG);
+
+            }
+            int totalMessages;
+            try {
+
+                totalMessages = Integer.parseInt(parameteres[TOTAL_MESSAGES_POSITION]);
+
+            }
+
+            catch (NumberFormatException e) {
+
+                throw new IllegalArgumentException(e.getMessage() + "\n" + TOTAL_MESSAGES_NOT_FOUND_ERROR_MSG);
+
+            }
+            int currentMessage;
+            try {
+
+                currentMessage = Integer.parseInt(parameteres[CURRENT_MESSAGE_POSITION]);
+
+            }
+
+            catch (NumberFormatException e) {
+
+                throw new IllegalArgumentException(e.getMessage() + "\n" + CURRENT_MESSAGE_NOT_FOUND_ERROR_MSG);
+
+            }
+            if (areValidParameters(actionType, parameteres[PARAM_POSITION], parameteres[EXTRA_POSITION],parameteres [PAYLOAD_POSITION])){
+
+                actionCommand = actionType;
+
+                param = parameteres[PARAM_POSITION];
+
+                extra = parameteres[EXTRA_POSITION];
+
+            }
+
+        }
+
+        else throw new IllegalArgumentException(FORMATTED_ACTION_NOT_FOUND_ERROR_MSG);
+
+    }
+
+    public boolean areValidParameters(int actionType, String param, String extra,String payload)
+    {
+     return Type.isValid(actionType) && (param.equals(DEFAULT_IGNORED) ^ Type.usesParam(actionType)) &&
+        (extra.equals(DEFAULT_IGNORED) ^ Type.usesParam(actionType))&&(payload.equals(DEFAULT_IGNORED) ^ Type.usesParam(actionType));
+    }
+    @Override
+    public boolean isValid() {
+        return currentPeerNode !=null && (
+                ((actionCommand == Type.INVITE || actionCommand == Type.ANSWER_INVITE || actionCommand==Type.PING ||
+                        actionCommand==Type.ANSWER_PING || actionCommand==Type.ANSWER_STORE || actionCommand==Type.FIND_NODE)
+                        && extra.equals(DEFAULT_IGNORED) && payload.equals(DEFAULT_IGNORED)&& !param.equals(DEFAULT_IGNORED))
+
+                || ((actionCommand==Type.FIND_VALUE || actionCommand==Type.ANSWER_FIND_VALUE ) &&
+                        payload.equals(DEFAULT_IGNORED) && !extra.equals(DEFAULT_IGNORED) &&!extra.equals(DEFAULT_IGNORED))
```

Comment:
Checking extra two times

Reply:
Yeah one of that was !param.equals(DEFAULT_IGNORED), i'll change it now